solution/index.js

```diff
@@ -0,0 +1,293 @@
+//general variables:
+const API = "https://json-bins.herokuapp.com/bin/614ad95a4021ac0e6c080c0d";
+const taskColumnIdArray = arrayOfColumnIds();
+const loader = loaderMaker();
+
+//website setup:
+setSectionsContent();
+
+/**
+ * local storage functions:
+ */
+//adds the input to specific array in local storage:
+function addToLocalStorage(key, input) {
+  if (!localStorage.getItem("tasks")) {
+    createLocalStorageDefaultItem();
+  }
+  const localStorageTasksObj = JSON.parse(localStorage.getItem("tasks"));
+  localStorageTasksObj[key].unshift(input);
+  localStorage.setItem("tasks", JSON.stringify(localStorageTasksObj));
+}
+
+//replaces the old text with edited text:
+function updateLocalStorage(key, indexOfCurrentValue, updatedValue = null) {
+  const localStorageTasksObj = JSON.parse(localStorage.getItem("tasks"));
+  const section = localStorageTasksObj[key];
+  if (updatedValue) {
+    section.splice(indexOfCurrentValue, 1, updatedValue);
+  } else {
+    section.splice(indexOfCurrentValue, 1);
+  }
+  localStorage.setItem("tasks", JSON.stringify(localStorageTasksObj));
+}
+
+/**
+ * event handlers functions:
+ */
+//adds the task to list on click
+function addButtonHandler(event) {
+  const relevantSection = event.path[1];
+  const input = relevantSection.children[2].value;
+  if (!input) {
+    alert("sorry... please enter your task");
+  } else {
+    const taskElm = createListElement(input);
+    relevantSection.children[1].insertBefore(
+      taskElm,
+      relevantSection.children[1].children[0]
+    ); //appends the new element to the top
+    addToLocalStorage(relevantSection.id, input);
+    event.target.parentNode.children[2].value = "";
+  }
+}
+
+//makes the content editable on double click, then adds blur event
+function doubleClickHandler(event) {
+  const target = event.target;
+  if (target.localName !== "li") {
+    return;
+  }
+  target.setAttribute("contenteditable", true);
+  target.focus();
+  target.addEventListener("blur", updateTaskHandler);
+}
+
+//makes the text uneditable and updates the local storage
+function updateTaskHandler(event) {
+  const relevantTaskElement = event.target;
+  relevantTaskElement.removeAttribute("contenteditable");
+  const columnName = event.path[2].id;
+  const indexOfOldText = getTaskElementIndex(columnName, relevantTaskElement);
+  const updatedText = relevantTaskElement.innerText;
+  relevantTaskElement.innerHTML = updatedText;
+  updateLocalStorage(columnName, indexOfOldText, updatedText);
+  if (!updatedText) {
+    relevantTaskElement.remove();
+  }
+}
+
+//sets unique name for relevant item, and adds event listener for keydown
+function mouseEnterHandler(event) {
+  const target = event.target;
+  target.setAttribute("name", "chosen");
+  document.body.addEventListener("keydown", keyDownHandler);
+}
+
+//resets every change the mouseenter did
+function mouseLeaveHandler(event) {
+  const target = event.target;
+  if (target.attributes[3].value === "chosen") {
```

Comment:
You can't rely on the attribute index. Someone might change the code, or the JS implementation or whatever, and your code will break.